webapp/src/ts/services/training-cards.service.ts

```diff
@@ -0,0 +1,124 @@
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+
+import { XmlFormsService } from '@mm-services/xml-forms.service';
+import { TrainingCardsComponent } from '@mm-modals/training-cards/training-cards.component';
+import { DbService } from '@mm-services/db.service';
+import { GlobalActions } from '@mm-actions/global';
+import { ModalService } from '@mm-modals/mm-modal/mm-modal';
+import { SessionService } from '@mm-services/session.service';
+import { RouteSnapshotService } from '@mm-services/route-snapshot.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TrainingCardsService {
+  private globalActions;
+  private readonly TRAINING_PREFIX = 'training:'
```

Comment:
Because this is used in multiple classes it would be better to be exposed somewhere rather than redefined everywhere it's used. That will help find other uses of it if we ever want to refactor it.